Xamarin.PropertyEditing.Mac/Controls/BasePointEditorControl.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections;
+using System.Diagnostics;
+using System.Drawing;
+using System.Reflection;
+using AppKit;
+using CoreGraphics;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal abstract class BasePointEditorControl<T> : PropertyEditorControl
+		where T : struct
+	{
+		protected NSTextView XLabel { get; set; }
+		protected NSTextField XEditor { get; set; }
+		protected NSTextView YLabel { get; set; }
+		protected NSTextField YEditor { get; set; }
+
+		internal new PropertyViewModel<T> ViewModel {
+			get { return (PropertyViewModel<T>)base.ViewModel; }
+			set { base.ViewModel = value; }
+		}
+
+		public BasePointEditorControl ()
+		{
+			var typeParameterType = typeof (T);
+
+			XLabel = new NSTextView ();
+
+			XEditor = new NSTextField ();
+			XEditor.BackgroundColor = NSColor.Clear;
+			XEditor.StringValue = string.Empty;
+			XEditor.Activated += (sender, e) => {
+				ViewModel.Value = (T)Activator.CreateInstance (typeParameterType, XEditor.FloatValue, YEditor.FloatValue);
```

Comment:
We're already going to the editor in the sub-classes, we should go _from_ it as well. This will crash for a Int32 based point (System.Drawing's).